lib/spack/llnl/util/link_tree.py

```diff
@@ -112,4 +126,7 @@ def unmerge(self, dest_root, **kwargs):
             elif os.path.exists(dest):
                 if not os.path.islink(dest):
                     raise ValueError("%s is not a link tree!" % dest)
-                os.remove(dest)
+                # only remove if src actually points to dest
```

Comment:
IIRC dest is actually the file/symlink in the view, so IMO it would be clearer if this read "if dest points to src" or perhaps even better is "only remove if dest is a hardlink/symlink to src" and it occurs to me that it would also be useful to explain how this can occur since it is not expected to be a common situation. So in all: 

"remove if dest is a hardlink/symlink to src; this will only be false if two packages are merged into a prefix and have a conflicting file"